draft-wang-ppm-dap-taskprov.md

```diff
@@ -56,76 +60,296 @@ collector). Thus it is up to each deployment of DAP to securely implement its
 own mechanism.
 
 This document describes a mechanism for configuring tasks that may be useful in
-many deployments.
-
-> TODO(wangshan) Describe the goals of this mechanism, including the deployment
-> scenario. Then describe how the extension works at a high level. I think the
-> intro should answer the following questions:
->
-> - Why does this mechanism need to be an extension (i.e., why are protocol
->   changes needed), as opposed to something that happens completely
->   out-of-band?
->
-> - Which of the sub-protocols are changed: upload, aggregate, collect?
-
+many deployments. The goal of this mechanism is to define a task provision
+method that utilizes just the upload channel and the metadata in the `Report`
+itself.
+
+At a high level, this extension asks client to include all the task
+configuration parameters it received out-of-band from task author, in the
+extension field of the `Report` it uploads to aggregators. The aggregators will
+create a DAP task upon receiving `Report` with such extension (or `ReportShare`
+in the case of helper.)
+
+By defining this mechanism as an extension, we can guarantee that for any
+deployments implementing this extension, all information required for
+aggregating a report is included in the report itself. There is no need for
+out-of-band task orchestration between leader and helpers, therefore making
+adoption of DAP easier. Since aggregators must create tasks using the same
+parameters client sent, this extension prevents task authors from giving clients
+and aggregators different task parameters in order to achieve reduced privacy
+guarantee, without requiring an out-of-band mechanism.
+
+This extension affects upload, aggregate and collect sub-protocols.
```

Comment:
This section should define the term "task author" (introduced by this draft) and elaborate on its role. We can do so in follow-up PR.

Reply:
It is defined in the Conventions and Definitions section:
> : The entity that defines the parameters of a task.

Reply:
Sure, but we definitely need some "flavor text" to gently introduce the reader to this concept. We can worry about this later though :)